src/app/_components/Specialists/FilterBase.js

```diff
@@ -0,0 +1,45 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import { FilterChip } from '@components/FilterChip';
+import PropTypes from 'prop-types';
+import { cn } from '@/utils/cn';
+
+export function FilterBase({ count, children, filterText }) {
+  const [opened, setOpened] = useState(false);
+  const dropdownRef = useRef(null);
+
+  const handleClickOutside = event => {
+    if (dropdownRef.current && !dropdownRef.current.contains(event.target) && opened) {
+      setOpened(false);
+      document.removeEventListener('click', handleClickOutside);
```

Comment:
Should you really remove Listener from this function?

Reply:
I think we need this listener only as long as dropdown is opened

Reply:
What happens if the user opens the dropdown menu again?

Reply:
Event listener added again. I see, no need to removing it, I'll change it

Reply:
Deleted it